### DISCUSSION TITLE
Python Packaging

### DISCUSSION THREAD

**Discussion by jpts** [brightio/penelope#16] [General]
:wave: Thanks for the tool, really like it.

To aid in packaging it in other projects, would you consider accepting a PR for a `pyproject.toml` file or similar? I understand the desire to have no external dependencies for the script, but it would aid packaging efforts elsewhere. Thanks!

**Comment by brightio** [comment]
Hi @jpts, I am glad you like it! I will need some time to familiarize myself with this concept and I will accept your pull request soon.

**Comment by brightio** [comment]
@jpts just merged your pull request. Thanks for that!

Can you please tell where this file can be used? I don't know much about python packaging. As I understand this file is a prerequisite to put this project to PyPI. Besides that where else can be used?

**Comment by jpts** [comment]
Just added another PR with some improvements/fixes :pray: 

Sure thing, I'm not exactly an expert, but it can be used anywhere this tool is packaged, not necessarily through PyPI. Part of the reason I submitted the request is that I submitted the package for inclusion in Nixpkgs ([PR](https://github.com/NixOS/nixpkgs/pull/240422)). You can see in the PR, the nix derivation (packaging instructions) pulls directly from GitHub releases. It is a well understood format, and integrates with existing tooling well.

Also, having the file may make it slightly easier for users to install as a CLI tool as they can clone the repo and run `pip install .` locally, at which point it gets installed in their PATH.

**Comment by brightio** [comment]
I see, thanks! Is the changing of the layout mandatory? I will have a quick look at the implications of that change and will respond.
